gettsim/benefits/alg2.py

```diff
@@ -221,50 +221,132 @@ def grossinc_alg2(household):
     )
 
 
-def e_anr_frei(household, params):
+def e_anr_frei_2005_01(household, params):
     """Calculate income not subject to transfer withdrawal for the household.
 
+    Legislation in force 2005-01-01 to 2005-09-30.
+
+    Determine the gross income that is not deducted. Withdrawal rates depend
+    on monthly earnings. § 30 SGB II."""
+
+    cols = ["m_wage", "ekanrefrei", "incometax", "soli", "svbeit"]
+    household.loc[:, cols] = household.groupby("pid")[cols].apply(
+        e_anr_frei_person_2005_01, params, params["a2eg3"]
+    )
+
+    return household
+
+
+def e_anr_frei_person_2005_01(person, params, a2eg3):
+    """Calculate income not subject to transfer withdrawal for each person.
+
+    Legislation in force 2005-01-01 to 2005-09-30."""
+
+    m_wage = person["m_wage"].iloc[0]
+
+    # Nettoquote
+    nq = alg2_2005_nq(person, params)
+
+    # Income not deducted
+    if m_wage <= params["a2eg1"]:
+        person["ekanrefrei"] = params["a2an1"] * nq * m_wage
+
+    elif params["a2eg1"] < m_wage <= params["a2eg2"]:
+        person["ekanrefrei"] = params["a2an1"] * nq * params["a2eg1"] + params[
+            "a2an2"
+        ] * nq * (m_wage - params["a2eg1"])
+
+    elif params["a2eg2"] < m_wage <= a2eg3:
+        person["ekanrefrei"] = (
+            params["a2an1"] * nq * params["a2eg1"]
+            + params["a2an2"] * nq * (params["a2eg2"] - params["a2eg1"])
+            + params["a2an3"] * nq * (m_wage - params["a2eg2"])
+        )
+
+    else:
+        person["ekanrefrei"] = (
+            params["a2an1"] * nq * params["a2eg1"]
+            + params["a2an2"] * nq * (params["a2eg2"] - params["a2eg1"])
+            + params["a2an3"] * nq * (a2eg3 - params["a2eg2"])
+        )
+
+    return person
+
+
+def alg2_2005_nq(person, params):
+    """Calculate Nettoquote
+
+    Quotienten von bereinigtem Nettoeinkommen und Bruttoeinkommen. § 3
+    Abs. 2 Alg II-V."""
+
+    # Bereinigtes monatliches Einkommen aus Erwerbstätigkeit. Nach § 11 Abs. 2 Nr. 1
+    # bis 5.
+    alg2_2005_bne = np.maximum(
```

Comment:
I think ``np.maximum`` is the wrong function. It is used to compare two arrays and find the maximum element-wise. It works for scalars, but in case something goes wrong, I assume the error message will be harder to understand. Thus, I would favor one of the following two
```
np.max((person..., 0))
np.clip(person..., 0, None)
```

Reply:
Done!